cmd/keys_rotate.go

```diff
@@ -0,0 +1,360 @@
+	os.Exit(1)
+}
+
+func loadScript(script string) (string, error) {
+	content := []byte(script)
+	tmpFile, err := ioutil.TempFile("/tmp", "*tmpScript")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temporary script: %w", err)
+	}
+	scriptPath := tmpFile.Name()
+	if _, err := tmpFile.Write(content); err != nil {
+		return "", fmt.Errorf("failed to write temporary script: %w", err)
+	}
+	if err := tmpFile.Close(); err != nil {
+		return "", fmt.Errorf("failed to save temporary script: %w", err)
+	}
+	if err := os.Chmod(scriptPath, 0777); err != nil {
+		return "", fmt.Errorf("failed to make script executable: %w", err)
+	}
+	return scriptPath, nil
+}
+
+func parseErrExitCode(err error) rotationExitCode {
+	if err == nil {
+		return RotationSuccess
+	}
+
+	if exitError, ok := err.(*exec.ExitError); ok {
+		exitCode := exitError.ExitCode()
+		code := rotationExitCode(exitCode)
+		for _, item := range rotationErrorExitCodes {
+			if item == code {
+				return item
+			}
+		}
+		logrus.Debugf("exit code unhandled: %r", code)
+	}
+	logrus.Debugf("exit code failed parse: %s", err)
+	return ErrUnknown
+}
+
+func runRotationScript(imageName string, sourcePath string, credentialsPath string) ([]byte, error) {
+	targetPath := "/tmp/tmp.py"
+	args := []string{
+		// base args
+		"run", "--rm",
+		// mount args
+		"-v", fmt.Sprintf("%s:/tmp/creds.tgz", credentialsPath),
+		"-v", fmt.Sprintf("%s:%s", sourcePath, targetPath),
+		// load args
+		imageName, targetPath,
+	}
+	command := exec.Command("docker", args...)
+	logrus.Debugf("executing command: %v\n", command)
+	return command.CombinedOutput()
+}
+
+func pullContainer(name string) error {
+	command := fmt.Sprintf("docker pull %s", name)
+	logrus.Debugf("pulling docker image %s\n", name)
+	if _, err := cli(command); err != nil {
+		logrus.Debugf("pulling image failed: %w", err)
+		return err
+	}
+	return nil
+}
+
+func verifyDocker() error {
+	logrus.Debugf("checking if docker is available")
+	if _, err := cli("which docker"); err != nil {
+		return fmt.Errorf("docker not available")
+	}
+	return nil
+}
+
+func copyFile(source string, target string) error {
+	from, err := os.Open(source)
+	if err != nil {
+		return fmt.Errorf("opening credentials path failed: %w", err)
+	}
+	defer from.Close()
+	fromStat, err := from.Stat()
+	if err != nil {
+		return fmt.Errorf("credentials details retrieval failed: %w", err)
+	}
+	to, err := os.OpenFile(target, os.O_RDWR|os.O_CREATE, 0666)
+	if err != nil {
+		return fmt.Errorf("opening backup path failed: %w", err)
+	}
+	defer to.Close()
+	if _, err = io.Copy(to, from); err != nil {
+		return err
+	}
+	if err = to.Chmod(fromStat.Mode()); err != nil {
+		return fmt.Errorf("backup permissions failed: %w", err)
+	}
+	return nil
+}
+
+func cli(inputs ...string) ([]byte, error) {
+	args := strings.Join(inputs, " && ")
+	cmd := exec.Command("/bin/sh", "-c", args)
+	logrus.Debugf("cli executing: %v", cmd)
+	return cmd.CombinedOutput()
```

Comment:
I'm guessing you had bigger plans for this originally. But you aren't making multple commands calls here requiring the "shell -c" and the "&&" stuff. I think you just need: `cmd := exec.Command(inputs)`

Reply:
lol that is exactly the case, fixed